beacon_node/beacon_chain/src/attestation_rewards.rs

```diff
@@ -147,18 +144,14 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             let weight = get_flag_weight(flag_index)
                 .map_err(|_| BeaconChainError::AttestationRewardsError)?;
 
-            let unslashed_participating_indices = participation_cache
-                .get_unslashed_participating_indices(flag_index, previous_epoch)?;
-
-            let unslashed_participating_balance =
-                unslashed_participating_indices
-                    .total_balance()
-                    .map_err(|_| BeaconChainError::AttestationRewardsError)?;
+            let unslashed_participating_balance = state
+                .progressive_balances_cache()
```

Comment:
this should be okay for now since the `progressive_balances_cache` is initialized by the call to `process_inactivity_updates_slow()` above.. but if you change that due to the concerns in the comment above then you'll need to consider if the `progressive_balances_cache` is initialized.

Reply:
added explicit initialisation & clarified that using `_slow` is OK here: https://github.com/sigp/lighthouse/pull/5386